src/oop/special_class_method/medium/q4/MainTest.java

```diff
@@ -0,0 +1,124 @@
+    public void setUp() throws Throwable {
+        String classString = "oop.special_class_method.medium.q4." + className;
+        this.testClass = new ObjectTest(classString);
+        String modifiedClassMessage =
+                "You have modified the provided portions of class " + className + ". Please revert them to the original state.";
+        assertTrue(testClass.hasField(attributeName, String.class), modifiedClassMessage);
+        assertTrue(testClass.hasModifier(attributeName, "private"), modifiedClassMessage);
+        String[] tests = new String[]{"Normal", "Exotic", "Green", "Flower", "Cactus"};
+        for (int i = 0; i < tests.length; i++) {
+            String value = tests[i];
+            Object[][] arguments = {
+                    {value, String.class}
+            };
+            Class<?>[] classes = {String.class};
+            Object testInstance = testClass.createInstance(arguments);
+            _assertEquals(value, testClass.getFieldValue(testInstance, attributeName), modifiedClassMessage);
+            assertTrue(testClass.hasModifier(classes, "public"), modifiedClassMessage);
+        }
+    }
+
+
+    private static Stream<String> stringInputProvider() {
+        return Stream.of(
+                "Normal", "Exotic", "Green", "Flower", "Cactus"
+        );
+    }
+
+
+    @ParameterizedTest
+    @MethodSource("stringInputProvider")
+    public void correctGetSizeMethod(String value) throws Throwable {
+        Object[][] arguments = {
+                {value, String.class}
+        };
+        Object testInstance = testClass.createInstance(arguments);
+        Object getMethodOutput = testClass.callMethod(getAttributeMethodName, testInstance);
+        assertTrue(testClass.hasModifier(getAttributeMethodName, null, "public"),
+                "Your " + getAttributeMethodName + " method does not have the correct visibility modifier.");
+        _assertEquals(value, getMethodOutput,
+                "Your " + getAttributeMethodName + " method does not return the value of the " + attributeName + " attribute.");
+    }
+
+    private static Stream<Arguments> twoStringsInputProvider() {
+        return Stream.of(
+                Arguments.of("Cactus", "Flower"),
+                Arguments.of("Green", "Normal"),
+                Arguments.of("Exotic", "Tree")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("twoStringsInputProvider")
+    public void correctSetSizeMethod(String initialValue, String updatedValue) throws Throwable {
+        Object[][] instantiationArguments = {
+                {initialValue, String.class}
+        };
+        Object testInstance = testClass.createInstance(instantiationArguments);
+        Object[][] setMethodArguments = {
+                {updatedValue, String.class}
+        };
+        Class<?>[] methodModifierClasses = {
+                String.class
+        };
+        testClass.callMethod(setAttributeMethodName, setMethodArguments, testInstance);
```

Comment:
we might want another assert before this to split the fail message for incorrect name and incorrect arguments